l10n_eu_nace/wizard/res_partner_industry_eu_nace_wizard.py

```diff
@@ -0,0 +1,183 @@
+        query = rf"""
+            PREFIX : <http://data.europa.eu/ux2/nace2/>
+            PREFIX cc: <http://creativecommons.org/ns#>
+            PREFIX dc: <http://purl.org/dc/elements/1.1/>
+            PREFIX dct: <http://purl.org/dc/terms/>
+            PREFIX eli: <http://data.europa.eu/eli/ontology#>
+            PREFIX euvoc: <http://publications.europa.eu/ontology/euvoc#>
+            PREFIX fn: <http://www.w3.org/2005/xpath-functions#>
+            PREFIX gn: <http://www.geonames.org/ontology#>
+            PREFIX grddl: <http://www.w3.org/2003/g/data-view#>
+            PREFIX iso-thes: <http://purl.org/iso25964/skos-thes#>
+            PREFIX luc: <http://www.ontotext.com/owlim/lucene#>
+            PREFIX nace21: <http://data.europa.eu/ux2/nace2.1/>
+            PREFIX owl: <http://www.w3.org/2002/07/owl#>
+            PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+            PREFIX rdf4j: <http://rdf4j.org/schema/rdf4j#>
+            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+            PREFIX sesame: <http://www.openrdf.org/schema/sesame#>
+            PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+            PREFIX skosxl: <http://www.w3.org/2008/05/skos-xl#>
+            PREFIX temporalDescriptions:
+            <http://publications.europa.eu/ontology/euvoc/temporalDescriptions#>
+            PREFIX vann: <http://purl.org/vocab/vann/>
+            PREFIX voaf: <http://purl.org/vocommons/voaf#>
+            PREFIX wgs: <http://www.w3.org/2003/01/geo/wgs84_pos#>
+            PREFIX xkos: <http://rdf-vocabulary.ddialliance.org/xkos#>
+            PREFIX xml: <http://www.w3.org/XML/1998/namespace>
+            PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+            SELECT ?LEVEL ?code ?parentCode {select_query} WHERE {{
+                ?s skos:inScheme :nace2 ;
+                {skos_query}
+                skos:notation ?code;
+                skos:broader ?parent;
+                ^skos:member ?Member.
+                ?parent skos:notation ?parentCode .
+                {filter_query}
+                FILTER (?Member = :sections || ?Member = :divisions  ||
+                ?Member = :groups || ?Member = :classes)
+                BIND (STR(?Member) AS ?NACELEVEL)
+                BIND (STRAFTER(?NACELEVEL, "/nace2/") AS ?LEVEL)
+                ?Member skos:prefLabel ?MemberLabel .
+                FILTER (LANG(?MemberLabel) = "en")
+            }} ORDER BY ?code
+        """
+        return query
+
+    def _create_nace_industry(self, nace_data, languages):
+        """Creates the partner industry records from the NACE data and adds it's
+        translations."""
+        nace_ids = self.env["res.partner.industry"]
+        nace_json = nace_data.json()
+        bindings = nace_json.get("results", {}).get("bindings", {})
+        for binding in bindings:
+            nace_code = binding.get("code", {}).get("value", "")
+            nace_id = self.env["res.partner.industry"].search(
```

Comment:
A search in a for can be a bottleneck. You could map them earlier.

Reply:
Done! Thanks for your review @EmilioPascual